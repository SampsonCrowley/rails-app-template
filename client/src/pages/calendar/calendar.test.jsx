import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AppointmentProvider from 'contexts/appointment';

import Calendar from './calendar';

describe('Pages - Calendar', () => {
  const div = document.createElement('div');
  const createCalendar = ({...props}) => {
    ReactDOM.render(
      <AppointmentProvider>
        <Calendar {...props} />
      </AppointmentProvider>
      , div);
    return div.querySelector('p')
  }

  it('is snapshotable', () => {
    const tree = renderer
      .create(
        <AppointmentProvider
          value={{
            appointmentState: {
              appointments: [],
              loaded: true
            },
            appointmentActions: {
              getAppointments: async () => []
            }
          }}
        >
          <Calendar />
        </AppointmentProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
  })
})
