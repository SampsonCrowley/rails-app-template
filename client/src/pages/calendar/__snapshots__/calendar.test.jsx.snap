// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages - Calendar is snapshotable 1`] = `
<div
  className="CalendarPage"
>
  <div
    style={
      Object {
        "display": "block",
        "width": "100%",
      }
    }
  >
    <h1
      className="text-center"
    >
      SUBMITTING...
    </h1>
    <div
      className="la-ball-atom la-vw-sm"
      style={
        Object {
          "color": "#00F",
          "marginLeft": "auto",
          "marginRight": "auto",
          "zIndex": 0,
        }
      }
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
